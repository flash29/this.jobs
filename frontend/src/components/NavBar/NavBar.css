.NavBar{
    display: flex;
    justify-content: space-around;
    background-color: black;
    color: aliceblue;
    height: 3.5em;
    align-items: center;
    font-size: 1.2em;
    font-weight: bold;
}