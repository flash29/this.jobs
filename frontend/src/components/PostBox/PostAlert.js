import React from 'react';
import "./PostAlert.css";
import { useNavigate } from "react-router-dom";

function PostAlert(props) {
    
    console.log(props);

    let navigate = useNavigate();

    const handleClick = () => {
        let path = props.url;
        navigate(path);
        props.setStatus(false);
    }

    return(
        props.trigger
        ?
        <div className = "popup">
            <div className = "popup-inner">
                <button className = "close-btn" onClick={handleClick}>Close</button>
                {props.children}
            </div>
        </div>
        :
        ""
    )
}

export default PostAlert;