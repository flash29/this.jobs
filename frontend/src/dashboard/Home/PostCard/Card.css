.CardType{
    display: flex;
    justify-content: center;
}

.fullClass{
    display: flex;
    justify-content: center;
    flex-direction: column;
    border-style: solid;   
    width: 25rem;
    border-radius: 10%;
    border: 0.4em solid rgb(243, 242, 242);
    border-radius: 16px;
    margin: 0.2em;
    padding: 0.2em;
}

.topCard{
    display: flex;
    justify-content: space-between;
    margin: 0.2em;
}

.Content{
    display: flex;
    justify-content: space-between;
    flex-direction: column;
    margin: 0.4em;
}

.textPost{
    display: flex;
    
}

.usernameDisplay{
    text-decoration: underline;
    font-weight: bold;
    font-size: 0.8em;
}

.usernameDisplay:hover{
    cursor: pointer;
    transform: scale(1.1);
    transition: 0.5s;
}

.timeDisplay{
    font-weight: lighter;
    font-size: 0.8em;
}

.likesCountDisplay{
    display: flex;
    opacity: 0.3;
    font-size: 0.8em;
    font-weight: bold;
    margin: 0.2em;
}

.PostTag{
    display: flex;
    font-weight:bold;
    font-size: 0.9em;
    background-color: rgba(255, 188, 170, 1);
    width: 9em;
    margin: 0.3em;
}

.likeAndCommentBoxes{
    display: flex;
    justify-content: space-around;
}

.likeBox{
    background-color: rgb(243, 242, 242);
    width: 12em;
    height: 1.8em;
    font-weight: bold;
    padding-top: 0.5em;
}

.likeBox:hover{
    cursor: pointer;
    text-decoration: underline;
    transform: scale(1.05);
    transition: 1s;
}

.commentBox{
    background-color: rgb(243, 242, 242);
    width: 12em;
    height: 1.8em;
    font-weight: bold;
    padding-top: 0.5em;
}
.commentBox:hover{
    cursor: pointer;
    text-decoration: underline;
    transform: scale(1.05);
    transition: 1s;
}

.AddComment{
    display: flex;
    padding-top: 1em;
    justify-content: space-between;
}

.CommentAddBox{
    display: flex;
    width: 22em;
    height: 2em;
    border: 0.4em solid rgb(243, 242, 242);
    border-radius: 16px;
}

.CommentAddBox:focus{
    outline: none;
    border: 0.4em solid rgb(0, 0, 0);
}

.CommentAddButton{
    background-color: rgb(243, 242, 242);
    font-weight: bold ;
    text-align: center;
    padding-top: 8px;
    border: 0.2em solid rgb(5, 2, 0);
    border-radius: 2em;
}

.CommentAddButton:hover{
    cursor: pointer;
    text-decoration: underline;
}