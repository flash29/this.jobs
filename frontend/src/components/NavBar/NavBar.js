import './NavBar.css';
import SearchBar from '../SearchBar/SearchBar'

function NavBar() {
  return (
    <div>
     <div className="NavBar">
       <div className="title">
          this.jobs
       </div>
       <div className="SearchBar">
          <SearchBar />
       </div>
       <div className="iconsHolder">
          Icons will be placed here
       </div>
      
     </div>
    </div>
  );
}

export default NavBar;