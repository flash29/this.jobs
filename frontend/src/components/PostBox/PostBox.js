import React from 'react';
import { Container, Button} from "react-bootstrap";
import "./PostBox.css";

import { useState} from 'react';
import sicon from "../../images/send-outline.svg";
import licon from "../../images/link-outline.svg";
import ficon from "../../images/document-attach-outline.svg";
import styled from 'styled-components';
import picon from "../../images/camera-outline.svg";

const Button1 = styled.button`
  position: relative;
  flex-wrap: wrap;
  flex-direction: row;
  border-radius: 10%;
  margin-right : 10px;
  width: 6%;
  height: 5%;
  background: transparent;
`;

function PostBox(props){

        const [postData, setPostData] = useState({ createdBy : '1', content : '', tag : '', attachments : ''});  

        const getBase64 = async (e) => {
            const file = e.target.files[0];
            const base64 = await convertBase64(file);
            return base64;
          }
        
        const convertBase64  = (file) => {
            return new Promise((resolve, reject) =>{
              const fileReader = new FileReader();
              fileReader.readAsDataURL(file);
              fileReader.onload = () => {
                resolve(fileReader.result);
              };
              fileReader.onerror = (error) => {
                console.log(error);
              };
            });
          }

        const hiddenFileInput = React.useRef(null);

        const handleClick = event => {
            hiddenFileInput.current.click();
        };

        function handleSubmit(){
            fetch('http://localhost:8080/post', {
                method : 'post', 
                headers:{'Content-type':'application/json'},
                body:JSON.stringify({postData})
            }).then(response => response.json()).then(data => {
                console.log(data);
                setPostData({ createdBy : '1', content : '', tag : '', attachments : ''});
            }).catch(error => console.log('error', error))
        }

        return(
            <Container className = "PostBox">
                <div >
                    <input className = "commentBox" placeholder = "What's on your mind?"  onClick = { (e) => setPostData({...postData, content : e.target.value})}/>
                </div>    
                <div className = "buttons">
                    <div className ="dropdown">
                            <select className = "dropbtn" onClick = {(e) => setPostData({...postData, tag : e.target.value})}>
                                <option value = "1">Select Tag</option> 
                                <option value = "Job-Recruitment">Job-Recruitment</option>
                                <option value = "Knowledge Sharing">Knowledge Sharing</option>
                                <option value = "Inspiration">Inspiration</option>
                                <option value = "Others">Others</option>    
                            </select>
                    </div>

                    <>
                        <Button1 onClick={handleClick}>
                            <img src = {picon} className = "images" alt = ""/>
                        </Button1>
                        <input
                            type="file"
                            ref = {hiddenFileInput}
                            style={{display: 'none'}}
                            onChange = { (e) => setPostData({...postData, attachments : getBase64(e)})}/>
                    </>


                    <>
                        <Button1 onClick={handleClick}>
                            <img src = {ficon} className = "images" alt = ""/>
                        </Button1>
                        <input
                            type="file"
                            ref={hiddenFileInput}
                            onChange = { (e) => setPostData({...postData, attachments : getBase64(e)})}
                            style={{display: 'none'}}
                        />
                    </>
                    <>
                        <Button1 onClick={handleClick}>
                            <img src = {licon} className = "images" alt = ""/>
                        </Button1>
                        <input
                            type="file"
                            ref={hiddenFileInput}
                            onChange = { (e) => setPostData({...postData, attachments : getBase64(e)})}
                            style={{display: 'none'}}
                        />
                    </>

                    <Button className = "icon" type = "submit" onClick={handleSubmit}>
                        <img src = {sicon} className = "images" alt = "" />
                    </Button>  
                </div>         
            </Container>    
        );    
    }

export default PostBox;
