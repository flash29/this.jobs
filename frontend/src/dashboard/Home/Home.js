import NavBar from '../../components/NavBar/NavBar'
import PostCard from './PostCard/PostCard';
import PostBox from '../../components/PostBox/PostBox';
import React, { useState, useEffect } from 'react';



function Home() {

  const [posts, setPosts] = useState();

  async function getPosts () {
      fetch('/feed/following/'+ sessionStorage.getItem('userid'), {
        //headers:{'Content-type':'application/json'},
        headers:{'Authorization' : 'Bearer ' + sessionStorage.getItem('token')},
        //body:JSON.stringify(postData),
    }).then(res => res.json())
        .then(res => setPosts(res) )
        .catch(error => console.log('error getting Feed'))
  }
  useEffect(() => {
    getPosts();
    console.log('posts', posts);
  }, []);


  return (
    <div className="App">
      <NavBar />
      <PostBox />
      <PostCard posts={posts}/>
    </div>
  );
}

export default Home;