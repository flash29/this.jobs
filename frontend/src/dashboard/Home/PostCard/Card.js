import './Card.css';
import React, { useState, useEffect } from 'react';
import CommentCard from '../CommentCard/CommentCard';
import { useNavigate } from "react-router-dom";

function Card(props) {
    let navigate = useNavigate();
    console.log("card", props);
    const [color, setColor] = useState('rgba(243, 242, 242)');
    const [likeChecker, setLikeChecker] = useState(false);
    const [commentContent, setCommentContent] = useState('');
    const [commentChecker, setCommentChecker] = useState(false);

    const {
        postId,
        tag,
        updatedAt,
        likes,
        createdBy,
        creatorId,
        createdAt,
        content,
        comments,
        attachments,
        LikesList
    } = props.postContent;

    console.log('postid', postId);

    console.log('comments', comments);


    let timeOfPost = '';

    function createDate(){
        let  currentTime= Math.floor(Date.now()/1000);
        let differenceDate = (currentTime - createdAt)/(60*60*24) ;
        if(differenceDate<1){
            // console.log('check proper time here', differenceDate*24 );
            differenceDate = differenceDate *24;
            // eslint-disable-next-line no-useless-concat
            timeOfPost = Math.floor(differenceDate) + ' ' + 'Hours Ago';
        }
        else{
            // eslint-disable-next-line no-useless-concat
            timeOfPost = Math.floor(differenceDate) + ' ' + 'Days Ago';
        }
    }

    let likesUsers = '';
    function createLikeDisplay(){
        if (likes===1){
            likesUsers = "A user has liked your post"
        }
        else{
            likesUsers = likes +" users have liked this post."
        }
    }
    createLikeDisplay();
    createDate();

    const onLikeClicked = (likeValue) =>{
        fetch('/updatelikes',{
            method:'put',
            headers:{'Content-type':'application/json'},
            body:JSON.stringify({
                postId: postId,
                user_id: createdBy,
                liked: likeValue
            })
        })
        .then(response=>response.json())
        .then(data=>{
            console.log(data);
        })
        .catch(error => console.log('error', error))
    }

    function likeClicked(){
        if(!likeChecker){
            setColor('rgba(255, 188, 170, 1)');
            setLikeChecker(true);
            onLikeClicked(true);
            console.log('post liked');
        }
        else{
            setColor('rgba(243, 242, 242)');
            setLikeChecker(false);
            onLikeClicked(false);
            console.log('post unliked');
        }
        
    }

    function onCommentContentChange(event){
        setCommentContent(event.target.value);

    }
    
    function commentClicked(){
        console.log(commentContent);
        fetch('/postcomment',{
            method:'post',
            headers:{'Content-type':'application/json'},
            body:JSON.stringify({
                    commentData : commentContent,
                    createdBy : createdBy,
                    post_id: postId
            })
        })
        .then(response=>response.json())
        .then(data=>{
            console.log(data);
            setCommentContent('');
            window.location.reload(false);
        })
        .catch(error => console.log('error', error))
    }

    function onCommentDisplayClicked(){
        if(!commentChecker){
            setCommentChecker(true);
        }
        else{
            setCommentChecker(false);
        }
    }

    const clickHandler = () => {
        console.log(props.postContent.creatorId)
        let path = "/settings/profile/"+props.postContent.creatorId;
        navigate(path);
    }

    return (
      <div className='CardType'>
          <div className='fullClass'>
            <div className="topCard">
                <div className='usernameDisplay' onClick={clickHandler} >{createdBy}</div> 
                <div className='timeDisplay'  >{timeOfPost}</div> 
            </div>
            <div className='PostTag'>#{tag}</div>
            <div className='Content'>
                <div className='textPost'>{content}</div>
                {
                    attachments 
                    ?
                    <>
                        <img src={attachments} alt={"Loading..."} />
                    </>
                    :
                    <>

                    </>
                }
            </div>
            <div className='likesCountDisplay'>
                {likesUsers}
            </div>
            <div className='likeAndCommentBoxes'>
                <div className='likeBox' 
                style={{backgroundColor:  color}}
                onClick={likeClicked}>Like</div>
                <div className='commentBox' onClick={onCommentDisplayClicked}>Comments</div>
            </div>
            {
                commentChecker 
                ?
                <>
                 {
                    comments.map((comment, i)=>{
                        console.log('comments', comment);
                        return (
                          <CommentCard 
                            key ={i}
                            commentContent = {comment}
                          />
                        );
                      })
                }
                </>
                :
                <></>
            }
            <div className='AddComment'>
                <input 
                type="text"
                placeholder = "Add a comment..."
                className='CommentAddBox'
                onChange={(e)=> onCommentContentChange(e) }
                value={commentContent}
                />
                <div className='CommentAddButton' onClick={()=>commentClicked()}> Comment </div>
            </div>
      </div>
      </div>  
      
    );
  }
  
  export default Card;