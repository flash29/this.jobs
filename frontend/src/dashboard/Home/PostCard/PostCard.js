import ServerLost from '../../../ErrorPage/ServerLost'
import FeedEmpty from '../../../ErrorPage/FeedEmpty'
import Card from './Card';

function PostCard(props) {
    console.log('props', props);
  return (
      <>
      {
          props.posts === undefined ?
            <ServerLost />
            :
            <>
            {
                props.posts.length === 0 ?
                <FeedEmpty />
                :
                <>
                {
                    props.posts.map((post, i)=>{
                        console.log('anslist', post);
                        return (
                          <Card 
                            key ={i}
                            postContent = {post}
                          />
                        );
                      })
                }
                </>
            }
            </>
      }
    </>
  );
}

export default PostCard;