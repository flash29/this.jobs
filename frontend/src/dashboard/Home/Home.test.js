import React from 'react';
import ReactDom from 'react-dom';
import { Router, BrowserRouter } from 'react-router-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import {createMemoryHistory} from 'history'
import Home from './Home';
import NavBar from '../../components/NavBar/NavBar'
import PostCard from './PostCard/PostCard';
import PostBox from '../../components/PostBox/PostBox';

import "@testing-library/jest-dom/extend-expect";
import { shallow, mount, configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer';

test('Homepage rendered right', () => {  
    const TextInputComponent = renderer.create(<BrowserRouter>
        <Home />
      </BrowserRouter>).toJSON();
    expect(TextInputComponent).toMatchSnapshot();
});

test('Navbar rendered right', () => {  
    const TextInputComponent = renderer.create(<BrowserRouter>
        <NavBar />
      </BrowserRouter>).toJSON();
    expect(TextInputComponent).toMatchSnapshot();
});

test('PostBox rendered right', () => {  
    const TextInputComponent = renderer.create(<BrowserRouter>
        <PostBox />
      </BrowserRouter>).toJSON();
    expect(TextInputComponent).toMatchSnapshot();
});

test('PostCard rendered right', () => {  
    const TextInputComponent = renderer.create(<BrowserRouter>
        <PostCard />
      </BrowserRouter>).toJSON();
    expect(TextInputComponent).toMatchSnapshot();
});


