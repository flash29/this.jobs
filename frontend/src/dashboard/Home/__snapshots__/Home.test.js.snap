// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage rendered right 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="NavBar"
    >
      <div
        className="title"
        onClick={[Function]}
      >
        this.jobs
      </div>
      <div
        className="SearchBar"
      >
        <div
          className="searchBarDisplay"
        >
          <input
            className="searchInput"
            id="searchInput"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Search"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="iconsHolder"
      >
        <a
          className="linkDisplay home "
          href="/home"
          onClick={[Function]}
        >
          <div
            className="iconsDisplay"
          >
            <img
              alt="Home"
              className="iconsImages "
              src="house_white_1.png"
            />
            <p
              className="iconLabel "
            >
              Home
            </p>
          </div>
        </a>
        <a
          className="linkDisplay connections "
          href="/connections"
          onClick={[Function]}
        >
          <div
            className="iconsDisplay"
          >
            <img
              alt="Home"
              className="iconsImages"
              src="people_white_1.png"
            />
            <p
              className="iconLabel"
            >
              Connections
            </p>
          </div>
        </a>
        <a
          className="linkDisplay jobs"
          href="/jobs"
          onClick={[Function]}
        >
          <div
            className="iconsDisplay"
          >
            <img
              alt="Home"
              className="iconsImages"
              src="work_white_1.png"
            />
            <p
              className="iconLabel"
            >
              Jobs
            </p>
          </div>
        </a>
        <a
          className="linkDisplay settings"
          href="/settings/profile/null"
          onClick={[Function]}
        >
          <div
            className="iconsDisplay"
          >
            <img
              alt="Home"
              className="iconsImages"
              src="settings_white_1.png"
            />
            <p
              className="iconLabel"
            >
              Settings
            </p>
          </div>
        </a>
        <a
          className="linkDisplay logout"
          href="/"
          onClick={[Function]}
        >
          <div
            className="iconsDisplay"
          >
            <img
              alt="Home"
              className="iconsImages"
              src="logout_white_1.png"
            />
            <p
              className="iconLabel"
            >
              Logout
            </p>
          </div>
        </a>
      </div>
    </div>
  </div>
  <div
    className="PostBox container"
  >
    <div
      className="box "
    >
      <h1
        className="titleToHover"
      >
        Create a Post!
      </h1>
      <div>
        <input
          className="textBox"
          onChange={[Function]}
          placeholder="What's on your mind?"
          type="text"
        />
      </div>
      <div
        className="buttons25"
      >
        <div
          className="dropdown"
        >
          <select
            className="dropbtn"
            onChange={[Function]}
          >
            <option
              id="SelectTag"
              value="1"
            >
              Select Tag
            </option>
            <option
              id="JobRecruitment"
              value="Job-Recruitment"
            >
              Job-Recruitment
            </option>
            <option
              id="KnowledgeSharing"
              value="Knowledge Sharing"
            >
              Knowledge Sharing
            </option>
            <option
              id="Inspiration"
              value="Inspiration"
            >
              Inspiration
            </option>
            <option
              id="Others"
              value="Others"
            >
              Others
            </option>
          </select>
        </div>
        <button
          className="sc-bdvvtL hvLuUM"
          onClick={[Function]}
        >
          <img
            alt=""
            className="images"
            src="camera-outline.svg"
          />
        </button>
        <input
          accept="image/x-png,image/gif,image/jpeg"
          id="inputPhotoPost"
          onChange={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          type="file"
        />
        <button
          className="postButton btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          Post
        </button>
      </div>
      <br />
    </div>
    
  </div>
  <div
    className="displayError"
  >
    <p>
      Looks like our server is not responding.
    </p>
    <p>
      Please try again.
    </p>
  </div>
</div>
`;

exports[`Navbar rendered right 1`] = `
<div>
  <div
    className="NavBar"
  >
    <div
      className="title"
      onClick={[Function]}
    >
      this.jobs
    </div>
    <div
      className="SearchBar"
    >
      <div
        className="searchBarDisplay"
      >
        <input
          className="searchInput"
          id="searchInput"
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="iconsHolder"
    >
      <a
        className="linkDisplay home "
        href="/home"
        onClick={[Function]}
      >
        <div
          className="iconsDisplay"
        >
          <img
            alt="Home"
            className="iconsImages "
            src="house_white_1.png"
          />
          <p
            className="iconLabel "
          >
            Home
          </p>
        </div>
      </a>
      <a
        className="linkDisplay connections "
        href="/connections"
        onClick={[Function]}
      >
        <div
          className="iconsDisplay"
        >
          <img
            alt="Home"
            className="iconsImages"
            src="people_white_1.png"
          />
          <p
            className="iconLabel"
          >
            Connections
          </p>
        </div>
      </a>
      <a
        className="linkDisplay jobs"
        href="/jobs"
        onClick={[Function]}
      >
        <div
          className="iconsDisplay"
        >
          <img
            alt="Home"
            className="iconsImages"
            src="work_white_1.png"
          />
          <p
            className="iconLabel"
          >
            Jobs
          </p>
        </div>
      </a>
      <a
        className="linkDisplay settings"
        href="/settings/profile/null"
        onClick={[Function]}
      >
        <div
          className="iconsDisplay"
        >
          <img
            alt="Home"
            className="iconsImages"
            src="settings_white_1.png"
          />
          <p
            className="iconLabel"
          >
            Settings
          </p>
        </div>
      </a>
      <a
        className="linkDisplay logout"
        href="/"
        onClick={[Function]}
      >
        <div
          className="iconsDisplay"
        >
          <img
            alt="Home"
            className="iconsImages"
            src="logout_white_1.png"
          />
          <p
            className="iconLabel"
          >
            Logout
          </p>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`PostBox rendered right 1`] = `
<div
  className="PostBox container"
>
  <div
    className="box "
  >
    <h1
      className="titleToHover"
    >
      Create a Post!
    </h1>
    <div>
      <input
        className="textBox"
        onChange={[Function]}
        placeholder="What's on your mind?"
        type="text"
      />
    </div>
    <div
      className="buttons25"
    >
      <div
        className="dropdown"
      >
        <select
          className="dropbtn"
          onChange={[Function]}
        >
          <option
            id="SelectTag"
            value="1"
          >
            Select Tag
          </option>
          <option
            id="JobRecruitment"
            value="Job-Recruitment"
          >
            Job-Recruitment
          </option>
          <option
            id="KnowledgeSharing"
            value="Knowledge Sharing"
          >
            Knowledge Sharing
          </option>
          <option
            id="Inspiration"
            value="Inspiration"
          >
            Inspiration
          </option>
          <option
            id="Others"
            value="Others"
          >
            Others
          </option>
        </select>
      </div>
      <button
        className="sc-bdvvtL hvLuUM"
        onClick={[Function]}
      >
        <img
          alt=""
          className="images"
          src="camera-outline.svg"
        />
      </button>
      <input
        accept="image/x-png,image/gif,image/jpeg"
        id="inputPhotoPost"
        onChange={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
        type="file"
      />
      <button
        className="postButton btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        Post
      </button>
    </div>
    <br />
  </div>
  
</div>
`;

exports[`PostCard rendered right 1`] = `
<div
  className="displayError"
>
  <p>
    Looks like our server is not responding.
  </p>
  <p>
    Please try again.
  </p>
</div>
`;
